addons/l10n_ar_stock/models/stock_picking.py

```diff
@@ -0,0 +1,91 @@
+        - Send allowed if: guide number exists.
+        """
+        for picking in self:
+            has_doc_type = bool(picking.picking_type_id.l10n_ar_document_type_id)
+            picking.l10n_ar_allow_generate_delivery_guide = (
+                picking.state == 'done' and has_doc_type and not picking.l10n_ar_delivery_guide_number
+            )
+            picking.l10n_ar_allow_send_delivery_guide = bool(picking.l10n_ar_delivery_guide_number)
+
+    # === Business Methods ===
+
+    def l10n_ar_action_create_delivery_guide(self):
+        """
+        Create the delivery guide number and store CAI data for the stock picking.
+        """
+        self.ensure_one()
+        self._check_company()
+
+        if not self.l10n_ar_delivery_guide_number:
+            picking_type = self.picking_type_id
+            self.l10n_ar_delivery_guide_number = f'{picking_type.l10n_ar_delivery_sequence_prefix}-{picking_type.l10n_ar_sequence_id.next_by_id()}'
+
+            self.l10n_ar_cai_data = {
+                'document_type_id': self.picking_type_id.l10n_ar_document_type_id.id,
+                'cai_authorization_code': self.picking_type_id.l10n_ar_cai_authorization_code,
+                'cai_expiration_date': self.picking_type_id.l10n_ar_cai_expiration_date.strftime('%Y-%m-%d'),
+                'sequence_number_start': self.picking_type_id.l10n_ar_sequence_number_start,
+                'sequence_number_end': self.picking_type_id.l10n_ar_sequence_number_end,
+            }
+
+    def l10n_ar_action_send_delivery_guide(self):
+        """
+        Send the delivery guide to the partner.
+        """
+        self.ensure_one()
+        if not self.partner_id.email:
+            raise UserError(_("The partner does not have an email address."))
+        template = self.env.ref('l10n_ar_stock.email_template_ar_remitos_delivery_guide')
+        pdf_action = self.env.ref('l10n_ar_stock.action_delivery_guide_report_pdf')
+        pdf_content, __ = self.env['ir.actions.report'].sudo()._render_qweb_pdf(pdf_action, self.id)
+        pdf_attachment = self.env['ir.attachment'].create({
+            'name': f'Remito - {self.l10n_ar_delivery_guide_number}.pdf',
+            'type': 'binary',
+            'datas': b64encode(pdf_content),
+            'res_model': self._name,
+            'res_id': self.id,
+            'mimetype': 'application/pdf',
+        })
+        composer = self.env['mail.compose.message'].with_context(
+            default_model='stock.picking',
+            active_model='stock.picking',
+            active_id=self.id,
+            default_res_ids=self.ids,
+            default_use_template=True,
+            default_template_id=template.id,
+            default_composition_mode='comment',
+            default_email_layout_xmlid='mail.mail_notification_layout_with_responsible_signature',
+            force_email=True,
+        ).create({
+            'attachment_ids': [(4, pdf_attachment.id)],
```

Comment:
Little nitpick :socks:, either use `Command.link` or in this case you can do:

```suggestion
            'attachment_ids': pdf_attachment.ids,
```